File: 02.code-run-debug/my-favorite-fruit.cpp
#include <iostream>
#include <string>
#include <time.h>
using namespace std;
struct timespec unit = { 0, 250000000L };
void w(int n, bool e = false, bool d = true) {
    for (int i = 0; i < n; i++) {
        if (d) { cout << "."; }
        nanosleep(&unit, NULL);
    }
    if (e) { cout << "!"; }
}
int main() {
    string my_favorite;
    cout << unitbuf;
    cout << "Input the favorite fruit: ";
    getline(cin, my_favorite);
    cout << "\nMy favorite fruit is";
    w(9);
    cout << " " << my_favorite;
    w(3, true);
    cout << "\n\n";
    return 0;
}

File: 02.code-run-debug/my-favorite-fruit.py
import time
def p(s): print(s, end="", flush=True)
def w(n, e=False, d=True):
    for i in range(n):
        if d: p(".")
        time.sleep(0.25)
    if e: p("!")
p("Input the favorite fruit: ")
myFavorite = input()
p("\nMy favorite fruit is")
w(9)
p(f" {myFavorite}")
w(3, True)
p("\n\n")
